onelogin_rules/onelogin_unusual_login.py

```diff
@@ -0,0 +1,60 @@
+import json
+import requests
+from panther_oss_helpers import get_string_set, put_string_set
+
+
+def rule(event):
+    # Pre-filter to save compute time where possible. event_type_id = 5 is login events.
+    if event.get('event_type_id') != 5 or event.get('ipaddr') is None:
+        return False
+
+    # Lookup geo-ip data via API call
+    url = 'https://ipinfo.io/' + event['ipaddr'] + '/geo'
+
+    # Skip API call if this is a unit test
+    if __name__ == 'PolicyApiTestingPolicy':
```

Comment:
Whatever works! :)

Reply:
I found out after writing this that it only works in the actual rules engine, not with the panther_analysis_tool :/ All the more reason to switch to the Panther public API when it's ready!